app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSitePermissionsCameraRobot.kt

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+/**
+ * Implementation of Robot Pattern for the settings Site Permissions Camera sub menu.
+ */
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.CoreMatchers
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.assertIsChecked
+import org.mozilla.fenix.helpers.click
+
+class SettingsSubMenuSitePermissionsCameraRobot {
+
+    fun verifyNavigationToolBarHeader() = assertNavigationToolBarHeader()
+
+    fun verifyassertAskToAllowRecommended() = assertAskToAllowRecommended()
+
+    fun verifyassertBlocked() = assertBlocked()
+
+    fun verifyCheckRadioButtonDefault() = assertCheckRadioButtonDefault()
+
+    fun verifyBlockedByAndroid() = assertBlockedByAndroid()
+
+    fun verifyToAllowIt() = assertToAllowIt()
+
+    fun verifyGotoAndroidSettings() = assertGotoAndroidSettings()
+
+    fun verifyTapPermissions() = assertTapPermissions()
+
+    fun verifyToggleCameraToON() = assertToggleCameraToON()
+
+    fun verifyGoToSettingsButton() = assertGoToSettingsButton()
+
+    fun verifySitePermissionsCameraSubMenuItems() {
+        verifyassertAskToAllowRecommended()
+        verifyassertBlocked()
+        verifyCheckRadioButtonDefault()
+        verifyBlockedByAndroid()
+        verifyToAllowIt()
+        verifyGotoAndroidSettings()
+        verifyTapPermissions()
+        verifyToggleCameraToON()
+        verifyGoToSettingsButton()
+    }
+
+    class Transition {
+        val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())!!
+
+        fun goBack(interact: SettingsSubMenuSitePermissionsRobot.() -> Unit): SettingsSubMenuSitePermissionsRobot.Transition {
+            mDevice.waitForIdle()

```

Comment:
Same as above..and for the rest of similar lines

Reply:
Ok